### PR TITLE
DB interface for mysql, sqlite3

### PR DESCRIPTION
**Pull request by divyenduz**
This can be initialization of https://github.com/mittsh/db-s3-backup/issues/4

Not very clean right now but later on adding any other DB will be simple implementation of DumpProtocol. That's the vision.

This is just a refactor as of now that retains the functionality.

### CODE REVIEW

File: db_s3_backup.py [hunk 1]
```diff
@@ -218,7 +174,7 @@ def delete_local_backups(dir_path, backup_prefix, backup_extension, verbose = Fa
 		default=False,
 	)
 	args = parser.parse_args()
-	
+
 	# Loads configuration file from JSON
 	try:
 		f=open(args.config_file, 'r')
```
Comment (mittsh) [inline db_s3_backup.py]: ideally one day we should turn all these old style `open()` to use context managers `with…:`

**Comment by mittsh** [comment]
I can't merge due to conflicts. Would you mind doing a rebase off this repo's `master` @divyenduz ?

**Comment by mittsh** [comment]
Btw, thanks for all the changes and work on this @divyenduz I'm glad to see that this little project is taking shape and is actually somewhat becoming useful for some! Many thanks 😄

**Comment by divyenduz** [comment]
hey @mittsh, thanks for the quick code review, appreciate it. I was not able to see any conflicts. I have synced this PR with master branch from the upstream repo as well

Can this be merged now ?

Thanks :)

**Comment by mittsh** [comment]
Yes absolutely! 👍  Thanks a lot for the PR.

**Pull request merged**